lib/foreman_tasks.rb

```diff
@@ -62,6 +62,25 @@ def self.delay(action, delay_options, *args)
     ForemanTasks::Task::DynflowTask.where(:external_id => result.id).first!
   end
 
+  # Chain a task to wait for prerequisite task(s) to finish before executing.
+  # The chained task remains 'scheduled' until all prerequisites reach 'stopped' state.
+  #
+  # @param plan_uuids [String, Array<String>] UUID(s) of prerequisite execution plan(s)
+  # @param action [Class] Action class to execute
+  # @param args Arguments to pass to the action
+  # @return [ForemanTasks::Task::DynflowTask] The chained task
+  def self.chain(plan_uuids, action, *args)
+    result = dynflow.world.chain(plan_uuids, action, *args)
+    ForemanTasks::Task.find_by(:external_id => result.id) ||
+      begin
+        delayed_plan = dynflow.world.persistence.load_delayed_plan(result.id)
+        execution_plan = dynflow.world.persistence.load_execution_plan(result.id)
+        ForemanTasks::Task::DynflowTask.new_for_execution_plan(execution_plan).tap do |task|
+          task.update_from_dynflow(execution_plan, delayed_plan)
+        end
+      end
```

Comment:
Is this really necessary? Chaining is "just" a special case of delay, for which we don't do this

Reply:
~~I thought I needed this for it to be visible as a scheduled task in the ForemanTasks browser, but I can revisit that.~~
Eh, scratch that. I can probably remove this. Let me just double check that there wasn't a corner case why we needed to update the task, I can't remember now.